scalameta/semantic/shared/src/main/scala/scala/meta/internal/semantic/Schema.scala

```diff
@@ -0,0 +1,97 @@
+package scala.meta
+package internal
+package semantic
+
+import java.net.URI
+import org.scalameta.data._
+import scala.{Seq => _}
+import scala.collection.immutable.Seq
+import scala.meta.io._
+import scala.meta.internal.semantic.{vfs => v}
+import scala.meta.internal.semantic.{schema => s}
+import scala.meta.{semantic => m}
+import scala.meta.inputs.{Input => mInput, Position => mPosition, Point => mPoint}
+
+package object schema {
+  @data class Database(entries: Seq[Attributes]) {
+    def toVfs(classpath: Classpath, sourcepath: Sourcepath): v.Database = {
+      if (classpath.shallow.isEmpty) sys.error("can't save semanticdb to an empty classpath")
+      val ventries = entries.map(entry => {
+        // TODO: Would it make sense to support multiclasspaths?
+        // One use-case for this would be in-place updates of semanticdb files.
+        val base = AbsolutePath(classpath.shallow.head)
+        val scalaNameOpt = sourcepath.relativize(AbsolutePath(entry.path).toURI)
```

Comment:
For #777, we will no longer need `sourcepath` in this method, because `entries` will already include a relative path for `sentry`.